ddprof-lib/src/main/cpp/symbols_linux.cpp

```diff
@@ -541,6 +633,16 @@ void Symbols::parseLibraries(CodeCacheArray *array, bool kernel_symbols) {
   // concurrent loading and unloading of shared libraries.
   // Without it, we may access memory of a library that is being unloaded.
   dl_iterate_phdr(parseLibrariesCallback, array);
+  TEST_LOG("Parsed %d libraries", array->count());
+}
+
+bool Symbols::isRootSymbol(const void* address) {
+  for (int i = 0; i < LAST_ROOT_SYMBOL_KIND; i++) {
+    if (ElfParser::_root_symbols[i] == (uintptr_t)address) {
```

Comment:
minor, the ordering could be smart (per code blob)

Reply:
Yeah, probably. I'll just let it dangling here. We are doing at most 5 comparisons which (I hope) compiler would just unroll anyway, and possibly vectorize.